streaming/src/java/org/apache/hive/streaming/RecordWriter.java

```diff
@@ -74,4 +92,27 @@
    * @return - set of partitions
    */
   Set<String> getPartitions();
+
+  /**
+   * Get the set of partitions that were use but that it wasn't possible
+   * to determine if they were used or not.
+   *
+   * @return - set of partitions
+   */
+  Set<String> getUndeterminedPartitions();
```

Comment:
You could use the same getPartitions() API. Maybe add a comment to say what it returns in what context.